cpp/common/src/codingstandards/cpp/Expr.qll

```diff
@@ -148,9 +148,17 @@ module MisraExpr {
   private predicate isCValue(Expr e) {
     not e.isConstant() and
     (
-      exists(ReturnStmt return | e = return.getExpr())
+      exists(ReturnStmt return |
+        e = return.getExpr() and
+        // Only return statements which are not explicitly casted are considered
+        not exists(Cast c | not c.isImplicit() and c.getExpr() = e)
```

Comment:
See topline comment.

Perhaps this should be something like,`e.getExplicitlyConverted() = return.getExpr()` ?

Reply:
Thanks, that's a good spot, I will make that change!